core/object/script_language_extension.h

```diff
@@ -670,9 +670,18 @@ class ScriptInstanceExtension : public ScriptInstance {
 			const GDExtensionPropertyInfo *pinfo = native_info->get_property_list_func(instance, &pcount);
 
 #ifdef TOOLS_ENABLED
-			Ref<Script> script = get_script();
-			if (script.is_valid() && pcount > 0) {
-				p_list->push_back(script->get_class_category());
+			if (native_info->get_class_category_func) {
+				GDExtensionPropertyInfo gdext_class_category;
+				if (native_info->get_class_category_func(instance, &gdext_class_category)) {
```

Comment:
We're losing the `pcount > 0` part of the original `if` statement.

I think we should probably wrap all the changed code in here in `if (pcount > 0) { ... }` because we don't want to include the class category at all, unless there's more than zero properties.